simln-lib/src/sim_node.rs

```diff
@@ -554,10 +561,12 @@ impl<T: SimNetwork> SimNode<T> {
                     "payment hash exists".to_string(),
                 ));
             },
-            Entry::Vacant(vacant) => {
-                vacant.insert(receiver);
-            },
-        }
+            Entry::Vacant(vacant) => vacant.insert(InFlightPayment {
+                track_payment_receiver: receiver,
+                path: route.paths[0].clone(), // TODO: MPP payments? we check in dispatch_payment
+                                              // should probably only pass a single path to dispatch
```

Comment:
Let's just send an error if we get more than one path? That way we restrict the API until we've got MPP support.

Wrote it up here: https://github.com/bitcoin-dev-project/sim-ln/issues/293
Happy for this to happen in a prefactor PR or just a preparatory commit in this PR.